auctions/dutch-auction/src/main.sw

```diff
@@ -0,0 +1,219 @@
+contract;
+
+dep abi;
+dep data_structures;
+dep errors;
+dep events;
+
+use std::{
+    address::Address,
+    assert::require,
+    block::height,
+    chain::auth::{AuthError, msg_sender},
+    context::{call_frames::msg_asset_id, msg_amount},
+    contract_id::ContractId,
+    identity::Identity,
+    logging::log,
+    result::*,
+    revert::revert,
+    storage::StorageMap,
+    token::{force_transfer_to_contract, transfer_to_output}
+};
+
+use abi::DutchAuction;
+use data_structures::Auction;
+use errors::{AuthorizationError, BidError, SetupError, TechnicalError, TimeError};
+use events::{AuctionCancelledEvent, CreatedAuctionEvent, WinningBidEvent};
+
+storage {
+    /// The Admin Address
+    admin: Identity,
+    /// Mapping an auction_id to its respective auction, allowing for multiple auctions to happen simultaneously
+    auctions: StorageMap<u64,
+    Auction>, /// Tracking how many auctions have been made till now
+    auction_count: u64,
+    /// Whether the constructor has been called
+    initialized: bool,
+}
+
+impl DutchAuction for Contract {
+    /// This function should be called right after being deployed, so the admin can be set
+    fn constructor(admin: Identity) {
+        require(!storage.initialized, AuthorizationError::CannotReinitialize);
+        storage.admin = admin;
+        storage.initialized = true;
+    }
```

Comment:
Constructor isn't needed anymore, nor any admin check. Delete anything to do with admin and initialization pls